lightning/src/ln/channelmanager.rs

```diff
@@ -4872,23 +4861,34 @@ where
 							node_id: channel.get_counterparty_node_id(),
 							msg: funding_locked,
 						});
-						if let Some(announcement_sigs) = self.get_announcement_sigs(channel) {
-							log_trace!(self.logger, "Sending funding_locked and announcement_signatures for {}", log_bytes!(channel.channel_id()));
-							pending_msg_events.push(events::MessageSendEvent::SendAnnouncementSignatures {
-								node_id: channel.get_counterparty_node_id(),
-								msg: announcement_sigs,
-							});
-						} else if channel.is_usable() {
-							log_trace!(self.logger, "Sending funding_locked WITHOUT announcement_signatures but with private channel_update for our counterparty on channel {}", log_bytes!(channel.channel_id()));
+						if channel.is_usable() {
+							log_trace!(self.logger, "Sending funding_locked with private initial channel_update for our counterparty on channel {}", log_bytes!(channel.channel_id()));
 							pending_msg_events.push(events::MessageSendEvent::SendChannelUpdate {
 								node_id: channel.get_counterparty_node_id(),
 								msg: self.get_channel_update_for_unicast(channel).unwrap(),
 							});
 						} else {
-							log_trace!(self.logger, "Sending funding_locked WITHOUT announcement_signatures for {}", log_bytes!(channel.channel_id()));
+							log_trace!(self.logger, "Sending funding_locked WITHOUT channel_update for {}", log_bytes!(channel.channel_id()));
 						}
 						short_to_id.insert(channel.get_short_channel_id().unwrap(), channel.channel_id());
 					}
+					if let Some(announcement_sigs) = announcement_sigs {
+						log_trace!(self.logger, "Sending announcement_signatures for channel {}", log_bytes!(channel.channel_id()));
+						pending_msg_events.push(events::MessageSendEvent::SendAnnouncementSignatures {
+							node_id: channel.get_counterparty_node_id(),
+							msg: announcement_sigs,
+						});
+						if let Some(height) = height_opt {
+							if let Some(announcement) = channel.get_signed_channel_announcement(self.get_our_node_id(), self.genesis_hash, height) {
```

Comment:
This seems to be a behavior change that could be worth documenting in the commit msg

Reply:
Wait, which part? The commit message explicitly states
```
    The spec actually requires we never send `announcement_signatures`
    (and, thus, `channel_announcement`s) until after six confirmations.
    However, we would happily have sent them prior to that as long as
    we exchange `funding_locked` messages with our countarparty.
```